bundles/org.openhab.binding.dlinksmarthome/src/main/java/org/openhab/binding/dlinksmarthome/internal/DLinkHNAP.java

```diff
@@ -0,0 +1,395 @@
+import org.eclipse.jetty.client.api.ContentResponse;
+import org.eclipse.jetty.client.api.Request;
+import org.eclipse.jetty.client.util.BytesContentProvider;
+import org.openhab.core.library.types.OnOffType;
+import org.openhab.core.library.types.QuantityType;
+import org.openhab.core.thing.ChannelUID;
+import org.openhab.core.thing.ThingStatus;
+import org.openhab.core.thing.ThingStatusDetail;
+import org.openhab.core.types.Command;
+import org.openhab.core.types.State;
+import org.openhab.core.types.UnDefType;
+import org.openhab.core.io.net.http.HttpClientFactory;
+import org.osgi.service.component.annotations.Activate;
+import org.osgi.service.component.annotations.Reference;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.NodeList;
+import org.xml.sax.SAXException;
+
+/**
+ * The {@link DLinkHNAP} class encapsulates the HNAP implementation for DLink thing handlers.
+ *
+ * @author Pascal Bies - Initial contribution
+ */
+public abstract class DLinkHNAP {
+    private Logger logger = LoggerFactory.getLogger(DLinkHNAP.class);
+    private String pin;
+    private String url;
+    private static final String USER = "admin";
+    private final HttpClient httpClient;
+
+    private ScheduledFuture<?> pollFuture;
+    private boolean invalidConfiguration = false;
+    Authentication authentication;
+    MessageFactory soapMessageFactory;
+    SOAPFactory soapFactory;
+    private static final int DETECT_POLL_S = 1;
+
+    public DLinkHNAP(final HttpClient httpClient) {
+        this.httpClient = httpClient;
+    }
+
+    private void poller() {
+        if (authenticate()) {
+            updateStatus(ThingStatus.ONLINE);
+            poll();
+        }
+    }
+
+    protected abstract void poll();
+
+    protected abstract void handleCommandAuthenticated(final ChannelUID channelUID, final Command command);
+
+    protected void updateStatus(ThingStatus status) {
+        updateStatus(status, ThingStatusDetail.NONE);
+    }
+
+    protected void updateStatus(ThingStatus status, ThingStatusDetail detail) {
+        updateStatus(status, detail, null);
+    }
+
+    protected abstract void updateStatus(ThingStatus status, ThingStatusDetail detail, final String message);;
+
+    private static class Authentication {
+        public Authentication(final String privateKey, final String cookie) {
+            this.privateKey = privateKey;
+            this.cookie = cookie;
+        }
+
+        public String privateKey;
+        public String cookie;
+    }
+
+    public void start(ScheduledExecutorService scheduler) {
+        try {
+            pollFuture = scheduler.scheduleWithFixedDelay(this::poller, 0, DETECT_POLL_S, TimeUnit.SECONDS);
+        } catch (Exception e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.HANDLER_INITIALIZING_ERROR,
+                    "Unexpected internal error.");
+            logger.error("Failed to start http client or scheduler.");
```

Comment:
Bindings should only log to error if something severe happened, like the detection of a bug in your code. This could be warn.

Additionally, you could remove the logging statement at all, as `updateStatus()` already does the logging.